src/control/lib/control/pool.go

```diff
@@ -0,0 +1,284 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package control
+
+import (
+	"context"
+	"os/user"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/daos-stack/daos/src/control/build"
+	"github.com/daos-stack/daos/src/control/common/proto/convert"
+	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+)
+
+const (
+	defaultSvcReps = 1
+)
+
+// checkUUID is a helper function for validating that the supplied
+// UUID string parses as a valid UUID.
+func checkUUID(uuidStr string) error {
+	_, err := uuid.Parse(uuidStr)
+	return errors.Wrapf(err, "invalid UUID %q", uuidStr)
+}
+
+// formatNameGroup converts system names to principal and if both user and group
+// are unspecified, takes effective user name and that user's primary group.
```

Comment:
Not your change, but we should probably revisit why we did it this way in the future. The daos_create_cont API doesn't require both params to be unspecified in order to pick up the effective user or effective group. It just fills in the unspecified param with the effective user or group.

Reply:
Sure, please create a ticket to address the tech debt and throw it in the 1.2 bucket if you think we'll get to it by then.

Reply:
https://jira.hpdd.intel.com/browse/DAOS-4547

Reply:
Yeah, I remember a conversation about this. I can't remember the use case idea for where we explicitly don't want an owner. The owner on a pool doesn't really matter much - it's a pretty mushy idea, compared to containers. It's just a way to have a default ACL that works for a specific user/group without too much admin interference.